src/hooks/auth.hook.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useCallback, useEffect } from 'react';
+
+const storageName = 'userData';
+
+export const useAuth = () => {
+    const [token, setToken] = useState(null);
+    const [ready, setReady] = useState(false);
+    const [userId, setUserId] = useState(null);
+
+    const login: any = useCallback((jwtToken: any, id: any) => {
+        setToken(jwtToken);
+        setUserId(id);
+
+        localStorage.setItem(
+            storageName,
+            JSON.stringify({
+                userId: id,
+                token: jwtToken,
+            })
+        );
+    }, []);
```

Comment:
Better to store this data in separate localStorage fields. In your application you should parse JSON everytime now. But this is a small mistake